### PR TITLE
Attempt to split models.py into files

### LINKED ISSUE #404: Organize models in a package
### Feature Inspiration

A large `models.py` file is hard to look through and update. 

### Feature Description

Break up `models.py` into a `models` package (see https://docs.djangoproject.com/en/4.0/topics/db/models/#organizing-models-in-a-package)

### Comment

Take care to communicate with other developers so they are aware of this incoming PR and can avoid making large changes to the `models.py` file during this refactor.

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

### PR DESCRIPTION
**Pull request by jcmatese**
## Summary Change Description

Split all the classes into separate files.
Attempted to get enough imports "fixed", to proceed to the "real" tests.

NOTE: This PR should be compared to #411 which possibly does the same, but with **_far fewer edits to existing files_**.

## Affected Issue Numbers

-resolves #404

## Code Review Notes

I deleted a `socket` symbol import (`TIPC_SUB_SERVICE`, but I can't seem to find it in the repository, now), because it was not found on my machine for some reason.  Double check importance?
Is there a better was to import multiple models /classes from distinctly named files?
What , if anything, should go in `__init__.py`?
Is there some way to provide a symbol table mapping/alias? (seems doubtful; @lparsons was thinking perhaps `__init__.py` could be leveraged)

## Checklist

- [ ] All issue requirements satisfied (or no linked issues)
- [ ] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [ ] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [ ] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [ ] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

File: DataRepo/models/compound.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    class Meta:
+        verbose_name = "compound"
+        verbose_name_plural = "compounds"
+        ordering = ["name"]
+
+    def __str__(self):
+        return str(self.name)
+
+class CompoundSynonym(models.Model):
+
```
Comment (jcmatese) [inline DataRepo/models/compound.py:91]: Every class has its own file, currently, except for `CompoundSynonym` which I just lumped into `compound.py`

File: DataRepo/views.py [hunk 2]
```diff
@@ -21,20 +21,34 @@
...
+from .models.msrun import MSRun
+from .models.peakdata import PeakData
+from .models.peakgroup import PeakGroup
+from .models.peakgroupset import PeakGroupSet
+from .models.protocol import Protocol
+from .models.sample import Sample
+from .models.study import Study
+from .models.tissue import Tissue
+from .models.tracerlabeledclass import TracerLabeledClass
+
```
Comment (jcmatese) [inline DataRepo/views.py:36]: I don't know if this is the "correct" way to style these imports, after file compartmentalization.

File: DataRepo/models/utilities.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
```
Comment (jcmatese) [inline DataRepo/models/utilities.py:3]: `models/utilities.py`  is a catch all for all the methods in the old `models.py` that were not in their own `class`.  We could name this whatever whatever, or organize it differently (had not considered it, when starting).

**Comment by jcmatese** [comment]
Suspect that the utilities `get_all_models` does not work correctly in this PR.
```
.venv) jcmatese@gen-matesembp:~/dev/tracebase(models-package) $ ./manage.py shell
Python 3.9.6 (v3.9.6:db3ff76da1, Jun 28 2021, 11:49:53) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.26.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from DataRepo.models.utilities import get_all_models

In [4]: lst = get_all_models()

In [5]: lst
Out[5]: [DataRepo.models.compound.CompoundSynonym, DataRepo.models.compound.Compound]
```

**Comment by hepcat72** [comment]
A convenient way to diff this PR from the branchpoint:

```
for f in DataRepo/models/*.py; do git -c color.diff=always diff HEAD^^^:DataRepo/models.py $f | grep -v ^.....\- -B 10 -A 10; done
```

**Review by hepcat72 — COMMENTED** [review]
I haven't investigated the tests WRT get_all_models.  While I don't have any issues aside from that, I do like the fact that Lance's version changes fewer files.

**Comment by jcmatese** [comment]
probably won't merge this (in favor of the other approach)

**Comment by lparsons** [comment]
@jcmatese Are you OK if we close this (I just merged #411 instead).

**Comment by jcmatese** [comment]
closed

**Pull request closed without merging**